### PR TITLE
`lokiexporter`: add complete log record to body

### LINKED ISSUE #6618: `lokiexporter`: add complete log record to body
At the moment, the Loki exporter ignores the log record properties like `Severity` and `TraceID`, causing these to be lost during the translation. While it makes sense to not add them as fields to the Loki record, they could be added as key/value pairs within the body itself, making them be recognized as fields by Loki.

### PR DESCRIPTION
**Pull request by jpkrohling**
Fixes #6618 by adding most fields, as well as the attributes and resource attributes not present in the config as part of the log body.

Signed-off-by: Juraci Paixão Kröhling <juraci@kroehling.de>

### CODE REVIEW

**Comment by jpkrohling** [comment]
This is how the log entry from the test looks like in Loki:

image

**Comment by nunojca** [comment]
This is really helpful to send logs in OpenTelemetry format through Loki Exporter to a backend such as Grafana Cloud. Thanks you.

**Review by gramidt — APPROVED** [review]
Thank you, @jpkrohling! LGTM

🚀

**Comment by pavankrish123** [comment]
LGTM!

File: exporter/lokiexporter/exporter.go [hunk 1]
```diff
@@ -227,14 +227,50 @@ func (l *lokiExporter) convertAttributesToLabels(attributes pdata.AttributeMap,
 	return ls
 }
 
-func convertLogBodyToEntry(lr pdata.LogRecord, res pdata.Resource) (*logproto.Entry, error) {
+func (l *lokiExporter) convertLogBodyToEntry(lr pdata.LogRecord, res pdata.Resource) (*logproto.Entry, error) {
+	body := lr.Body().StringVal()
+
+	if len(lr.Name()) > 0 {
+		body = fmt.Sprintf("name=%s %s", lr.Name(), body)
```
Comment (anuraaga) [inline exporter/lokiexporter/exporter.go:234]: While I'm not deeply familiar with its performance in Go in most languages repeated String generation like this is very expensive.

How about using a `strings.Builder`?

https://pkg.go.dev/strings#Builder

Comment (jpkrohling) [reply exporter/lokiexporter/exporter.go:234]: Done!

File: exporter/lokiexporter/exporter.go [hunk 2]
```diff
@@ -227,14 +229,68 @@ func (l *lokiExporter) convertAttributesToLabels(attributes pdata.AttributeMap,
...
+
 	return &logproto.Entry{
 		Timestamp: time.Unix(0, int64(lr.Timestamp())),
-		Line:      lr.Body().StringVal(),
+		Line:      b.String(),
 	}, nil
 }
 
-func convertLogToJSONEntry(lr pdata.LogRecord, res pdata.Resource) (*logproto.Entry, error) {
+func (l *lokiExporter) convertLogToJSONEntry(lr pdata.LogRecord, res pdata.Resource) (*logproto.Entry, error) {
```
Comment (bogdandrutu) [inline exporter/lokiexporter/exporter.go:293]: Unnecessary change?

Comment (jpkrohling) [reply exporter/lokiexporter/exporter.go:293]: Not exactly, the non-JSON method got changed, I thought it would be better for consistency to have them both part of the lokiExporter.

**Comment by jpkrohling** [comment]
@anuraaga, @bogdandrutu, could you please review this one again?

**Review by anuraaga — APPROVED** [review]

File: exporter/lokiexporter/exporter.go (hunk 2 above)
Comment (jpkrohling) [reply exporter/lokiexporter/exporter.go:293]: @bogdandrutu, I'm merging this PR as is, but if you think this is a concern, let me know and I'll revert this.
